### DISCUSSION TITLE
The query speed suddenly slowed down

### DISCUSSION THREAD

**Discussion by ingale726** [milvus-io/milvus#21488] [Q&A and General discussion]
```
_DIM = 768  # dimension of vector
_INDEX_FILE_SIZE = 32  # max file size of stored index
search_param = {'nprobe': 16}
index_param = {
    'nlist': 2048
}
IndexType = IVF_FLAT

gpu:
  enable: true
  cache_size: 1GB
  gpu_search_threshold: 0
  search_devices:
    - gpu0
  build_index_devices:
    - gpu0
```
I turned on the GPU query, Single GPU graphics card

I was inserting data, I tested the query speed under 1 second, the data volume at that time was around 800,000, but when the data increased to more than 1 million, the query speed changed to 10 seconds, what happened? It seems to become a query using the CPU

**Comment by yhmo** [comment]
Dynamic data is not suitable for GPU searching. With inserting data, the GPU need to load new generated segments into GPU memory,  time waste in data copying between CPU memory and GPU memory.

**Reply by ingale726** [reply]
I have stopped inserting and then executing
`milvus.flush([collection_name]) `
but the query speed is still over 10 seconds. I found a cache_size setting in the document. I guess 2G is not enough. I set it to 10 gigabytes,and set _INDEX_FILE_SIZE = 2048.
I'll try the effect now and reply later

**Comment by ingale726** [comment]
```
Search time: 20.83630871772766
Search time: 21.145814657211304
Search time: 20.811245679855347
Search time: 20.703389167785645
```
I have completed the insertion, and now the retrieval speed is 20 seconds. When the data volume is about 1.4 million, the retrieval speed is still less than one second, and when it exceeds 1.4 million, the speed becomes more than 10 seconds. 
now I've inserted about 3.04 million pieces of data
I didn't set other parameters, I don't know why
@yhmo

**Comment by yhmo** [comment]
Typically, 3 million vectors, search performance should be 10~50ms.
Not sure what is the problem, log can be helpful.

**Comment by ingale726** [comment]
in the log,I see a setting, 
`cache.enable: false`
Its default state is false
This setting is not added in the startup configuration file 
`conf/server_config.yaml`

```
gpu:
  enable: true
  cache_size: 10GB
  cache.enable: true
  gpu_search_threshold: 1
  search_devices:
    - gpu0
  build_index_devices:
    - gpu0
```
I added this option to my startup profile and set it to true, and my continuous query dropped to about 0.1 seconds
I am confused, why is it not visible in the startup configuration file, I think the reason for my slow query is here
@yhmo

**Reply by yhmo** [reply]
https://github.com/milvus-io/milvus/issues/5142
